restcomm/restcomm.http/src/main/java/org/mobicents/servlet/restcomm/http/GeolocationEndpoint.java

```diff
@@ -0,0 +1,752 @@
+        final GsonBuilder builder = new GsonBuilder();
+        builder.registerTypeAdapter(Geolocation.class, converter);
+        builder.setPrettyPrinting();
+        gson = builder.create();
+        xstream = new XStream();
+        xstream.alias("RestcommResponse", RestCommResponse.class);
+        xstream.registerConverter(converter);
+        xstream.registerConverter(new ClientListConverter(configuration));
+        xstream.registerConverter(new RestCommResponseConverter(configuration));
+    }
+
+    public Response putGeolocation(final String accountSid, final MultivaluedMap<String, String> data,
+            GeolocationType geolocationType, final MediaType responseType) {
+        try {
+            secure(accountsDao.getAccount(accountSid), "RestComm:Create:Geolocation");
+            secureLevelControl(accountsDao, accountSid, null);
+        } catch (final AuthorizationException exception) {
+            return status(UNAUTHORIZED).build();
+        }
+        try {
+            if (!geolocationType.toString().equalsIgnoreCase("Immediate")
+                    && !geolocationType.toString().equalsIgnoreCase("Notification")) {
+                validateGeolocationType(data);
+            }
+        } catch (final NullPointerException exception) {
+            return status(BAD_REQUEST).entity(exception.getMessage()).build();
+        }
+
+        Geolocation geolocation = createFrom(new Sid(accountSid), data, geolocationType);
+
+        if (geolocation.getResponseStatus() != null && geolocation.getResponseStatus().equalsIgnoreCase("rejected")) {
+            System.out.println("Geolocation ResponseStatus rejected or unauthorized for that Sid");
+            if (APPLICATION_XML_TYPE == responseType) {
+                final RestCommResponse response = new RestCommResponse(geolocation);
+                return ok(xstream.toXML(response), APPLICATION_XML).build();
+            } else if (APPLICATION_JSON_TYPE == responseType) {
+                return ok(gson.toJson(geolocation), APPLICATION_JSON).build();
+            } else {
+                return null;
+            }
+        } else {
+
+            dao.addGeolocation(geolocation);
+
+            if (APPLICATION_XML_TYPE == responseType) {
+                final RestCommResponse response = new RestCommResponse(geolocation);
+                return ok(xstream.toXML(response), APPLICATION_XML).build();
+            } else if (APPLICATION_JSON_TYPE == responseType) {
+                return ok(gson.toJson(geolocation), APPLICATION_JSON).build();
+            } else {
+                return null;
+            }
+        }
+    }
+
+    private Geolocation createFrom(final Sid accountSid, final MultivaluedMap<String, String> data,
+            Geolocation.GeolocationType glType) {
+        // *** Validation of not null and specific formatted parameters *** //
+        // *** Source can not be null ***/
+        try {
```

Comment:
I think all this validation code should be isolated in another place, this method is getting big.... Also some parts of the validation may be designed to reuse code